hail/hail/src/is/hail/expr/ir/lowering/ExecuteRelational.scala

```diff
@@ -84,9 +84,8 @@ object ExecuteRelational {
         TableValueIntermediate(TableValue(ctx, typ, globals, rvd))
       case TableMapGlobals(child, newGlobals) =>
         TableValueIntermediate(recur(child).asTableValue(ctx).mapGlobals(newGlobals))
-      case TableMapRows(child, newRow) =>
-        val extracted = agg.Extract(newRow, r.requirednessAnalysis, isScan = true)
-        TableValueIntermediate(recur(child).asTableValue(ctx).mapRows(extracted))
+      case ir: TableMapRows =>
+        TableStageIntermediate(ctx.backend.tableToTableStage(ctx, ir, r))
```

Comment:
This forces the entire `child` pipeline to be lowered, which isn't what we want here. See the `TableJoin` case as an example; we want to factor out the `TableMapRows` lowering implementation to be called directly from both the table lowering pass and here.